pkg/nsx/services/inventory/pod.go

```diff
@@ -44,7 +44,7 @@ func (s *InventoryService) initContainerApplicationInstance(clusterId string) er
 		}
 		applicationInstances, _, err := s.NSXClient.NsxApiClient.ContainerApplicationsApi.ListContainerApplicationInstances(context.Background(), opts)
 		if err != nil {
-			return fmt.Errorf("failed to retrieve container application instances err: %w", err)
+			return fmt.Errorf("failed to retrieve ContainerApplicationInstances err: %w", err)
 		}
 		for _, applicationInstance := range applicationInstances.Results {
 			if applicationInstance.ContainerProjectId == "" {
```

Comment:
Not related to this PR, just notice here is a stalePods slice saves containerinventory.ContainerApplicationInstance instead of *containerinventory.ContainerApplicationInstance, not sure if you would like to update it to pointer similar to pendingAdd  and pendingDelete

https://github.com/vmware-tanzu/nsx-operator/blob/0a71934245b9edd1181e4f36a4b8809323aaf75b/pkg/nsx/services/inventory/pod.go#L53

Reply:
change it to pointer